src/app/highlight.directive.ts

```diff
@@ -0,0 +1,28 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+
+@Directive({
+  selector: '[appHighlight]'
+})
+export class HighlightDirective {
+
+  constructor(private el: ElementRef) { }
+
+  @Input('appHighlight') highlightColor: string;
+
+
+  @HostListener('mouseenter') onMouseEnter() {
+    this.highlight(this.highlightColor || 'red');
+
+    console.log('hello');
+  }
+
+
+  @HostListener('mouseleave') onMouseLeave() {
+    this.highlight(null);
+  }
+
+  private highlight(color: string) {
+    this.el.nativeElement.style.backgroundColor = color;
```

Comment:
Instead of directly changing the style use Renderer2 API. (https://angular.io/api/core/Renderer2#setstyle)
As an Angular good practices, we don't apply HTML changes directly to the element.

Example:
`constructor(private renderer: Renderer2, private el: ElementRef) {}

ngOnInit() {
  this.renderer.setStyle(
    this.el.nativeElement,
    'background-color',
    'olive'
  );
}`

Reply:
Noted and resolved.